packages/contracts/utils/etherscan.ts

```diff
@@ -77,3 +86,410 @@ export const runTaskWithRetry = async (
     }
   }
 };
+
+/**
+ * @returns All contracts with the name containing 'Proxy' from the deployments stored on disk
+ */
+export const fetchProxies = (deployments: AllDeployments) =>
+  Object.entries(deployments)
+    .map(([name, deploy]) => ({name, address: deploy.address.toLowerCase()}))
+    .filter(deployment => deployment.name.match(/Proxy$/g))
+    // Naming conventions is to call the base contract XXXProxy, but HH Deploy
+    // will suffix _Proxy (leading to XXXProxy_Proxy).
+    // This second filter removes that noise as they're the same contract anyway.
+    .filter(deployment => !deployment.name.match(/Proxy\_Proxy$/g));
+
+/**
+ * @returns All contracts with the name containing 'Implementation' from the deployments stored on disk
+ */
```

Comment:
What happens with implementations that we don't deploy directly, but that are deployed indirectly with another contract (i.e., the `implementation` being part of a `PluginSetup`)?

Reply:
On second thought, this happens for `PluginSetup`s only that will be moved to a separate repository and handled by a separate deploy script anyway.
I think it is fine to ignore this special case